ipq40xx: add support for 8devices Jalapeno

robimarko opened the pull request:
This patch adds support for 8devices Jalapeno.

Specification:
* QCA IPQ4018, Quad core ARM v7 Cortex A7 717MHz
* 256 MB of DDR3 RAM
* 8 MB of SPI NOR flash
* 128 MB of Winbond SPI NAND flash
* WLAN1: Qualcomm Atheros QCA4018 2.4GHz 802.11bgn 2:2x2
requires special BDF in QCA4019/hw1.0/board-2.bin with:
bus=ahb,bmi-chip-id=0,bmi-board-id=16,variant=8devices-Jalapeno
* WLAN2: Qualcomm Atheros QCA4018 5GHz 802.11a/n/ac 2:2x2
requires special BDF in QCA4019/hw1.0/board-2.bin with:
bus=ahb,bmi-chip-id=0,bmi-board-id=17,variant=8devices-Jalapeno
* ETH: Qualcomm Atheros QCA8072 Gigabit Switch (1 x LAN, 1 x WAN)
phy@mdio3:
Label: eth0
gmac0
phy@mdio4:
Label: eth1
gmac1

Installation instructions:
Since boards ship with old version of LEDE installation is simple.
Just use sysupgrade -n -F sysupgrade.bin

Syuspgrade needs to be forced since OpenWRT uses DT detection in recent
releases.

If you get error that FIT configuration is not found during boot it is
due to older U-boot used on your board.
That is because 8devices used custom FIT configuration partition name
as they internally had v1 and v2 boards.
Only v2 boards are sold so now they are shipping boards with never
U-boot using generic config@1 FIT partition name.

Also for old uboot it is possible to force loading config@1 by changing
uboot environment:

setenv boot5 'bootm 0x84000000#config@1’
saveenv

Signed-off-by: Robert Marko <robimarko@gmail.com>

blogic commented:
@robimarko patch looks fine but the whole alphabetical ordering it broken, please fix it.

robimarko commented:
@blogic Thanks for the review,it should be resolved now.

pepe2k commented on target/linux/ipq806x/config-4.9:
```diff
@@ -313,6 +313,8 @@ CONFIG_MTD_M25P80=y
 CONFIG_MTD_NAND_ECC=y
 CONFIG_MTD_NAND_QCOM=y
+CONFIG_MTD_SPINAND_MT29F=y
+CONFIG_MTD_SPINAND_ONDIEECC=y
```
Did you add that manually?

robimarko replied:
Sorry, I don't understand what do you mean by that?

pepe2k replied:
You didn't use `make kernel_menuconfig CONFIG_TARGET=target` and put these lines by hand instead, am I right? :)

robimarko replied:
Yes, that is correct.

pepe2k replied:
> Yes, that is correct.

Then it's wrong, you broke alphabetical order here.

robimarko replied:
@pepe2k Thank you,I did not know how to use make kernel_menuconfig CONFIG_TARGET=target so far.
Now it should be correct

blogic commented on target/linux/ipq806x/base-files/etc/board.d/02_network:
```diff
@@ -52,6 +52,9 @@ zyxel,nbg6817)
 	ucidef_set_interface_macaddr "lan" "$(macaddr_add $hw_mac_addr 2)"
 	ucidef_set_interface_macaddr "wan" "$(macaddr_add $hw_mac_addr 3)"
 	;;
+8dev,jalapeno)
```
when doing alphanumeric ordering you put numbers before letters ... even if the name suggests otherwise.

robimarko commented:
@blogic Sorting done, also rebased on master to resolve conflicts after GL Inet gl-b1300 was merged

pepe2k commented on target/linux/ipq806x/files-4.9/arch/arm/boot/dts/qcom-ipq4019-jalapeno.dts:
```diff
@@ -0,0 +1,351 @@
+			status = "ok";
+		};
+
+		usb3_ss_phy: ssphy@9a000 {
```
You have many nodes duplicated...

robimarko replied:
I reused a lot of 8devices DTS and that is part of it,was not paying attention to it.
Removed 3 copies of USB nodes.

pepe2k commented:
@robimarko is this board supported in ath10k-firmware upstream (board-2.bin data)?

pepe2k commented:
@robimarko did you see my previous comment? Your PR is probably missing important chunk for IPQ40xx based board - calibration data (board-2.bin).

robimarko commented:
@pepe2k Yes,I saw it.
From all I can see it is supported,8devices does not use any custom board-2.bin in their LEDE sources.
I am now building this latest version to test it out,and I can provide you with bootlog confirming that ath10k detects everything fine.

robimarko commented:
Just finished and here it new bootlog
https://gist.github.com/robimarko/8a583a964deffcb8d297276a4f370e05

You can see that BMI ID is recognized and OTP and everything else is loaded.

ecsv commented:
@robimarko Just as extra info: The problem on these QCA wifi components is not that it isn't recognized but that it doesn't work as (good as) expected. I don't know the radio details of this board - so it depends on what @8devices got when they prepared (calibrated) the BDFs (board data files) for their board. Maybe they actually ended up with the exact same data as the AP.DK01 - but HW manufacturers often seem to need special BDFs. On previous HW models, everything was encoded in the ART partition but the newer HW models use a different strategy. Part of the data is in the BDFs (all stored in these board-2.bin container files) and the rest (pre-cal) is stored in the ART partition of each device. These are then combined by the OTP (and maybe even more modified when it checks the efuses) and given to the firmware.

Let us assume for a moment that it isn't using the same BDFs as the AP.DK01. Then the correct BDFs have to be submitted to Kalle Valo. The process can be found in:

* official documentation
  - https://wireless.wiki.kernel.org/en/users/drivers/ath10k/boardfiles
* how I upstreamed the A42 (everything should actually be in one mail when submitting the BDFs)
  - https://www.mail-archive.com/ath10k@lists.infradead.org/msg07569.html (initial mail before the official guide existed)
  - https://www.mail-archive.com/ath10k@lists.infradead.org/msg07642.html (answers to the questions from the guide)

The new version of ath10k-firmware has then to be integrated in OpenWrt. And then you have to modify the DTS to provide the correct qcom,ath10k-calibration-variant for each wifi device. There is also the ipq-wifi package which can be used when the processing of the BDF submission takes too long. But the package itself also makes a couple of problems. So it would be (imho) better to try to get the BDFs upstreamed as early as possible.

We could ask @8devices (especially @mantas-p) for details.

You can send me the 0:ART partition (copy it using dd) and I can check whether the pre-cal data looks like they were made with the pristine QCA BDFs. You should have my mail address because you contacted me when you had problems to get the board to boot.

mantas-p commented:
@ecsv Thanks for a ping and the detailed explanation on board-2.bin

We're using slightly modified AP.DK01 boarddata so submission to upstream would be needed. I've uploaded boarddata files for reference: https://github.com/mantas-p/files/tree/master/jalapeno_bdf

robimarko commented:
@ecsv Thanks for taking time to explain everything.
I knew that there was a lot of custom BDF-s used by it looked like this board was supported by upstream since BMI and everything was recognized and even 8devices repo has no mention or trace of code pointing to use of custom BMI.
I can still provide you with ART dump if you want,how does one recognize BMI from it?

I was also following your way of upstreaming BDF for A42 before.
@mantas-p Thanks for the BDF-s,I will try to upstream them and until then I think ipq-wifi will be needed.

Just gotta figure out how to use two binary-s for it

ecsv commented:
Thanks for the fast response. I've just prepared the upstreaming mail and will sent them to Kalle after the mailinglist server reacts again.

@robimarko no need with the provded BDFs.

robimarko commented:
Awesome,thank you.

ecsv commented:
@robimarko You already have to add the qcom,ath10k-calibration-variant because the board-8dev_jalapeno.bin is already using it.

----
Btw. I was asked via mail how the board-8dev_jalapeno.bin was created.

You have to clone https://github.com/qca/qca-swiss-army-knife and then write a  board-2.json file to map the data files to the ID strings

    [
        {
            "data": "8DEV3000_JALAPENO_2G_boarddata_0_v1.6.bin", 
            "names": [
                "bus=ahb,bmi-chip-id=0,bmi-board-id=16,variant=8devices-Jalapeno"
            ]
        }, 
        {
            "data": "8DEV3000_JALAPENO_5G_boarddata_0_v1.6.bin", 
            "names": [
                "bus=ahb,bmi-chip-id=0,bmi-board-id=17,variant=8devices-Jalapeno"
            ]
        }
    ]

And then you can create the board-2.bin when you run following in the folder with all the data files (and board-2.json):

    $(PATH_TO_QCA_SWISS_ARMY_KNIFE)/tools/scripts/ath10k/ath10k-bdencoder -c board-2.json

The board-8dev_jalapeno.bin is just the renamed board-2.bin which was created in this step.

robimarko commented:
@ecsv Thank you for all the help.
Will add qcom,ath10k-calibration-variant to DTS

robimarko commented:
PR updated to include custom board-2.bin.
Runtime tested.

robimarko commented:
@blogic Any chance of merging this or I have to wait until ipq40xx split?

ecsv commented:
@robimarko The split was pushed and you have to move your changes to the ipq40xx target. And they also want to have some changes in the DT. You can see them in https://github.com/openwrt/openwrt/commit/d0e962140403eefbe70002ee4349ade1d8c119be

* Move every node which has a label in the included DTSI to the top level (below the main/root DT node)
* Order entries in the DT according to their addresses
* Make sure that the device tree uses the correct name for the SoC: qcom-ipq4018-jalapeno.dts
* check that the labels on the ethernet ports are somehow documented together with their phy_mdio_addr (will become especially relevant later with the DSA driver)
  - maybe check whether the labels on the board currently match what is assigned in Linux (will change later with the DSA driver but could be interesting to know)
  - Mathias Kresin prefers when eth1 is the WAN port under IPQ40xx
* merge spi_0_pins's pinmux and pinconf in a node called pin
* merge spi_0_pins's pinmux_cs and pinconf_cs in a node called pin_cs
* remove disabled nodes (like mx25l25635e) which are not present in the included DTSI files
* Add the qcom,ath10k-calibration-variant to your wifi0/wifi1 

I hope that I have mentioned everything relevant in this list

robimarko committed:
ipq-wifi: Add 8devices Jalapeno

Add custom board-2.bin for 8devices Jalapeno.
Upstreaming is in progress.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
@ecsv Changed as per your list.
It should be fine now

mkresin commented on package/boot/uboot-envtools/files/ipq40xx:
```diff
@@ -31,6 +31,7 @@ ubootenv_mtdinfo () {
 }
 
 case "$board" in
+8dev,jalapeno |\
```
why do you need access to uboot env? I don't see anything changing something in the env. Furthermore, the uboot env write protected.

I'm quite sure it can be dropped.

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+	};
+
+	soc {
+
```
drop this blank line

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+		};
+
+		ess-psgmii@98000 {
+			status = "ok";
```
Please use `status = "okay";` as it is the only valid value listed in the [Devicetree Speciﬁcation, Release v0.2 - 2.3.4 status](https://www.devicetree.org/specifications/).

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+			reg = <0x4a1000 0x4>;
+		};
+
+		pinctrl@1000000 {
```
This node has a label in the qcom-ipq4019.dtsi. You can (and should) reference the pinctrl via `&tlmm`

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+	soc {
+
+		mdio@90000 {
+			status = "ok";
```
```
pinctrl-0 = <&mdio_pins>;
pinctrl-names = "default";
```

is missing here

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0>;
+				compatible = "mx25l1606e", "n25q128a11";
```
only `compatible = "jedec,spi-nor";` please

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+				#size-cells = <0>;
+				reg = <0>;
+				compatible = "mx25l1606e", "n25q128a11";
+				linux,modalias = "m25p80", "mx25l1606e", "n25q128a11";
```
This property is obsolete since kernel 4.4. Drop it.

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+			status = "ok";
+		};
+
+		cryptobam: dma@8e04000 {
```
please reference this node via `&cryptobam`.

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+			status = "ok";
+		};
+
+		wifi@a000000 {
```
please reference this node via `&wifi0`.

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+			qcom,ath10k-calibration-variant = "8devices-Jalapeno";
+		};
+
+		wifi@a800000 {
```
please reference this node via `&wifi1`.

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+
+		edma@c080000 {
+			status = "okay";
+			gmac0 {
```
reference this node via &gmac0.

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+				qcom,phy_mdio_addr = <3>;
+				vlan_tag = <1 0x10>;
+			};
+			gmac1 {
```
reference this node via &gmac1.

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,275 @@
+			status = "ok";
+		};
+
+		blsp_dma: dma@7884000 {
```
reference this node via `&blsp_dma`.

mkresin commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -32,6 +32,22 @@ define Device/UbiFit
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
+	DEVICE_DTS := qcom-ipq4018-jalapeno
+	KERNEL_LOADADDR := 0x80208000
```
Can be dropped, as it is now added to the `Device/Default`

mkresin commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -32,6 +32,22 @@ define Device/UbiFit
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	BOARD_NAME := jalapeno
+	SUPPORTED_DEVICES := 8dev,jalapeno
```
can be dropped, it is set to the same value in `Device/Default`.

mkresin commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -32,6 +32,22 @@ define Device/UbiFit
+	KERNEL_SIZE := 4048k
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	BOARD_NAME := jalapeno
```
This shouldn't be required since quite some time.

mkresin commented on target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:
```diff
@@ -119,6 +119,7 @@ case "$FIRMWARE" in
 	avm,fritzbox-4040)
 		/usr/bin/fritz_cal_extract -i 1 -s 0x400 -e 0x207 -l 12064 -o /lib/firmware/$FIRMWARE $(find_mtd_chardev "urlader_config")
 		;;
+	8dev,jalapeno |\
```
move the whole block to the top, to keep alphabetical order. Same for the other block in this file.

robimarko commented:
@mkresin Your comments were adressed

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,277 @@
+		#size-cells = <0x1>;
+		ranges;
+
+		rsvd1@87E00000 {
```
lowercase unit address please. the dtc compiler show warn you about the the unit address and reg mismatch.

mkresin commented on target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts:
```diff
@@ -0,0 +1,277 @@
+
+&wifi0 {
+	status = "okay";
+	qcom,ath10k-calibration-variant = "8devices-Jalapeno";
```
as far as I know, your board.bin wasn't yet accepted upstream yet. If I'm right, please make it a comment. Same for the other wifi node.

ecsv replied:
The board-2.bin in ipq-wifi is already using the variant strings:

    $ strings board-8dev_jalapeno.bin|grep bus
    bus=ahb,bmi-chip-id=0,bmi-board-id=16,variant=8devices-Jalapenom
    bus=ahb,bmi-chip-id=0,bmi-board-id=17,variant=8devices-Jalapenom

So giving it in the DTS is required (and nice do have when it is merged in the official board-2.bin)

mkresin replied:
Thanks a lot for the explanation! Quite obvious that it need to be kept as it is now.

mkresin commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -32,6 +32,19 @@ define Device/UbiFit
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
+	DEVICE_DTS := qcom-ipq4018-jalapeno
+	KERNEL_INSTALL := 1
```
I don't see an explanation why the raw kernel is required as extra file. Copy/Paste leftover?

robimarko replied:
Yes, It was copied straight from 8devices branch.

mkresin commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -32,6 +32,19 @@ define Device/UbiFit
+	$(call Device/UbiFit)
+	DEVICE_DTS := qcom-ipq4018-jalapeno
+	KERNEL_INSTALL := 1
+	KERNEL_SIZE := 4048k
```
Where does the kernel size limit comes from? It looks like the kernel is a ubi volume. In that case there is usually not size restriction for the kernel.

robimarko replied:
Same as above.
Yes,everything is on ubi so there is no actuall limit

mkresin reviewed (changes requested):
Only minor things found this time and it already looks very good.

robimarko commented:
@mkresin Last changes you requested are pushed.
Runtime tested off course.

mkresin commented:
@ecsv does this PR needs any changes wrt #810 or is it ready to be merged?

ecsv commented:
> @ecsv does this PR needs any changes wrt #810 or is it ready to be merged?

Here are the things which I've noticed while comparing this PR with the changes from #810:

* the rsvd1 might be splitted into smem and tz (see https://github.com/openwrt/openwrt/pull/810/commits/30fb78174aab1155ef24f18b75a06e7ce7a8b935 )
  - it should be checked whether this is actually possible or whether the u-boot from 8devices modifies the rsvd1 memory
  - it might be interesting to know whether tz_apps@87b80000 (0x87b80000 0x280000) is also required (@mantas-p can you help here)
* cs-gpios could use the define GPIO_ACTIVE_HIGH as the last parameter (instead of the 0)

robimarko commented:
@ecsv I can use GPIO_ACTIVE_HIGH but reserved memory is completely copy/paste from 8devices branch
https://github.com/8devices/lede-8dev/blob/feed-8dev/target/linux/ipq806x/files-4.9/arch/arm/boot/dts/qcom-ipq40xx-ap.jalapeno.dts#L26

ecsv commented:
> @ecsv I can use GPIO_ACTIVE_HIGH but reserved memory is completely copy/paste from 8devices branch

This is the reason why I've mentioned @mantas-p . We already know that their DT is far from perfect and required modifications. So it seemed to be a good idea to ask @8devices directly to get the information:

* Is their u-boot modifying the reserved-memory regions?
* Is their u-boot dependent on the rsvd1 reserved-memory region in the DTS?
* Which regions must be protected from modifications because they are used by non-Linux components.
  - Must only the tz (TrustZone; <0x87e80000 0x180000>) + smem (<0x87e00000 0x080000>) regions be protected?
  - What about the tz_apps regions (<0x87b80000 0x280000>)?
  - Are they using different mappings (regions)?
  - Are any other regions mentioned in 156f2df879fd5afbe1aee2e9d37d81e36195a59b ("ipq806x: ipq4019: add ap-dk01.1-c1 board support") relevant for their board?
  - Do they have knowledge about which component(s) would access cpu_context_dump or cnss_debug and why/whether they must be protected under Linux?
  - Do they know why apps_bl is protected under Linux in the default DTs of the AP.DK boards? (u-boot already gave the control over to Linux - so it doesn't seem to be necessary to be protected in any way)
  - Same question for apps_bl also seems to be relevant for sbl (secondary bootloader)

robimarko commented:
GPIO_ACTIVE_HIGH applied and conflicts after #810 resolved.
Today I discovered a weird issue,after couple of days  sitting turned off I found SSH to be disabled and Luci would claim wrong username/password but I did not set one

mantas-p commented:
@ecsv 

1) Split `rsvd1` to `smem+tz` is OK, U-Boot does not modify DTS here.
2) We don't use any Trustzone apps and not reserving that memory had no adverse effects, but since we don't have source of tz.bin we can't be 100% sure TZ is not touching anything there, so reserving `tz_apps` might be a good idea.
3) `cnss_debug` and `cpu_context_dump` are related to WiFi subsytem crash dump/debug. It seems to work only with proprietary driver anyway. Support for them is disabled in our u-boot.
4) `apps_bl` and `sbl` is memory used by U-boot and 'Secondary boot loader' and need to save them is also related to crash dump feature, so in OpenWRT/non-debug case there is no point having them.

ecsv commented:
Thanks for the fast feedback. @robimarko maybe you can try copying the reserved memory from A42 or WPJ428 and check that everything still works.

ratkaj commented:
Tested these patches. My Jalapeno works, no issues that i see so far.

robimarko commented:
@ecsv Copied from A42 and it is working.
I think that this is ready to be merged.

ecsv commented:
Do the ethernet ports have some kind of labels? Here is how I will document these for A62: https://github.com/ecsv/openwrt/commit/6044d05f175d318f4b22c77a522fd80f7769f442

robimarko commented:
No,but I can add those

robimarko commented:
@ecsv Added,they match the ones in A62 exactly

ecsv commented:
Just searched for pictures and they are not labeled like the A62/A42 on the the official product pages: https://www.8devices.com/media/products/jalapeno/Jalapeno_Development.jpg

robimarko commented:
They are labeled as eth0 and eth1 on the pcb.
Does not make any difference.

On 2 April 2018 at 17:05, Sven Eckelmann <notifications@github.com> wrote:

> Just searched for pictures and they are not labeled like the A62/A42 on
> the the official product pages: https://www.8devices.com/
> media/products/jalapeno/Jalapeno_Development.jpg
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/713#issuecomment-377949944>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABTDsPSZJJ-tY71gm0lBrTLhp7mEypFFks5tkj44gaJpZM4R4SSC>
> .
>

robimarko commented:
@mkresin Any chance of merging this before the fork for 18.05?

mkresin commented:
At least the usb related nodes are still defined in your dts instead of referenced.

ecsv commented:
> They are labeled as eth0 and eth1 on the pcb.
> Does not make any difference.

It makes a difference. The upcoming(tm) DSA driver will have labels. And knowing the right label for each port can therefore be relevant.

> At least the usb related nodes are still defined in your dts instead of referenced.

He is talking about the nodes with the label usb3_ss_phy, usb3_hs_phy, and usb2_hs_phy from qcom-ipq4019.dtsi. usb2@60f8800 and usb3@8af8800 don't have a label

robimarko committed:
ipq40xx: add support for 8devices Jalapeno

This patch adds support for 8devices Jalapeno.

Specification:
QCA IPQ4018, Quad core ARM v7 Cortex A7 717MHz
256 MB of DDR3 RAM
8 MB of SPI NOR flash
128 MB of Winbond SPI NAND flash
WLAN1: Qualcomm Atheros QCA4018 2.4GHz 802.11bgn 2:2x2
requires special BDF in QCA4019/hw1.0/board-2.bin with:
bus=ahb,bmi-chip-id=0,bmi-board-id=16,variant=8devices-Jalapeno
WLAN2: Qualcomm Atheros QCA4018 5GHz 802.11a/n/ac 2:2x2
requires special BDF in QCA4019/hw1.0/board-2.bin with:
bus=ahb,bmi-chip-id=0,bmi-board-id=17,variant=8devices-Jalapeno
ETH: Qualcomm Atheros QCA8072 Gigabit Switch (1 x LAN, 1 x WAN)
phy@mdio3:
Label: eth0
gmac0
phy@mdio4:
Label: eth1
gmac1

Installation instructions:
Since boards ship with old version of LEDE installation is simple.
Just use sysupgrade -n -F sysupgrade.bin

Syuspgrade needs to be forced since OpenWRT uses DT detection in recent
releases.

If you get error that FIT configuration is not found during boot it is
due to older U-boot used on your board.
That is because 8devices used custom FIT configuration partition name
as they internally had v1 and v2 boards.
Only v2 boards are sold so now they are shipping boards with never
U-boot using generic config@1 FIT partition name.

Also for old uboot it is possible to force loading config@1 by changing
uboot environment:

setenv boot5 'bootm 0x84000000#config@1’
saveenv

Signed-off-by: Robert Marko <robimarko@gmail.com>

mkresin commented:
merged, thanks!
